File: src/main/java/com/example/probation/service/TokenService.java
package com.example.probation.service;

import com.example.probation.event.OnRegistrationCompleteEvent;
import com.example.probation.core.entity.User;
import com.example.probation.core.entity.VerificationToken;

import java.util.Optional;

public interface TokenService {
    void saveNewToken(VerificationToken token);

    Optional<VerificationToken> findByToken(String token);

    Optional<User> getUserByToken(String token);

    void confirmRegistration(OnRegistrationCompleteEvent event);
}

File: src/main/kotlin/com/example/probation/service/TokenService.kt
package com.example.probation.service

import com.example.probation.core.entity.User
import com.example.probation.core.entity.VerificationToken

interface TokenService {
    fun saveNewToken(token: VerificationToken)

    fun findByToken(token: String): VerificationToken?

    fun getUserByToken(token: String): User?

    fun confirmRegistration(user: User)

    fun deleteTokenByUser(user: User)
}
